.circleci/config.yml

```diff
@@ -316,12 +316,62 @@ jobs:
       - run:
           name: Generate SQL content
           command: |
+            # Check if the generated-sql branch can simply be re-used and SQL generation can be skipped.
+            git clone -b generated-sql git@github.com:mozilla/bigquery-etl ~/generated-sql
+            cd ~/generated-sql
+
+            # Commits on generated-sql are tagged with the sha1 hash of the commit on main that triggered the SQL generation
+            export LATEST_TAG_GENERATED_SQL=`git describe --tags --abbrev=0`
+            LATEST_TAG_GENERATED_SQL=${LATEST_TAG_GENERATED_SQL/c-/""}
+            echo "Latest tag on generated-sql branch: c-$LATEST_TAG_GENERATED_SQL"
+
+            # Get the state in which main was in when the generated-sql branch was created and pushed
+            cd ~
+            git clone -b main git@github.com:mozilla/bigquery-etl ~/bigquery-etl-main
+            cd ~/bigquery-etl-main
+            export LATEST_COMMIT_ON_MAIN=`git rev-parse --short HEAD`
+            git checkout $LATEST_TAG_GENERATED_SQL
+            echo "Latest commit on main: $LATEST_COMMIT_ON_MAIN"
+
+            cd ~/project
+            export GENERATED_SQL_CHANGED=false
+
+            # If this task runs when merging into main, then always run the SQL generation
+            if [ "$CIRCLE_BRANCH" = main ]; then
+              GENERATED_SQL_CHANGED=true
+            fi
+
+            # Check if this PR removes a file. SQL generation needs to be re-run because
+            # simply merging the changes with the generated-sql branch would result in files being part of generated DAGs
+            # and other artifacts that were actually removed
+            if [[ $(git diff --no-index --name-only --diff-filter=D ~/bigquery-etl-main/sql sql) ]]; then
+              echo "SQL file was removed; re-generate SQL"
+              GENERATED_SQL_CHANGED=true
+            fi
+
+            # Check if this PR changes SQL generators
+            if [[ $(diff -qr --no-dereference sql_generators ~/bigquery-etl-main/sql_generators) ]]; then
+              GENERATED_SQL_CHANGED=true
```

Comment:
Since the SQL generators use utils and other parts of bigquery-etl it's possible that changes outside of the `sql_generators` dir could result in different `generated-sql` content. I _think_ it's unlikely missing one of those would result in anything too hard to recover from but something to be aware of. Might be a good idea to have an escape hatch to set `GENERATED_SQL_CHANGED` to `true` on your PR on demand